guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/fork/ForkRepositoryViewImpl.java

```diff
@@ -0,0 +1,232 @@
+
+    @UiField
+    TextBox targetOrganizationalUnitTextBox;
+
+    @UiField
+    FormLabel sourceOrganizationalUnitLabel;
+
+    @UiField
+    FormLabel targetOrganizationalUnitLabel;
+
+    @UiField
+    BaseModal popup;
+
+    @UiField
+    FormGroup managedReposiotryGroup;
+
+    @UiField
+    CheckBox managedRepository;
+
+    @UiHandler("fork")
+    public void onForkClick( final ClickEvent e ) {
+        presenter.handleForkClick();
+    }
+
+    @UiHandler("cancel")
+    public void onCancelClick( final ClickEvent e ) {
+        presenter.handleCancelClick();
+    }
+
+    @Override
+    public void init( final Presenter presenter,
+                      final boolean isOuMandatory ) {
+        this.presenter = presenter;
+        setBody( uiBinder.createAndBindUi( this ) );
+
+        setTitle( CoreConstants.INSTANCE.ForkRepository() );
+
+        sourceOrganizationalUnitTextBox.addValueChangeHandler( e -> {
+            sourceOrganizationalUnitGroup.setValidationState( ValidationState.NONE );
+        } );
+
+        targetOrganizationalUnitTextBox.addValueChangeHandler( e -> {
+            targetOrganizationalUnitGroup.setValidationState( ValidationState.NONE );
+        } );
+
+    }
+
+    @Override
+    public String getSourceSelectedOrganizationalUnit() {
+        return sourceOrganizationalUnitTextBox.getValue();
+    }
+
+    @Override
+    public void setSourceOrganizationalUnitGroupType( final ValidationState state ) {
+        sourceOrganizationalUnitGroup.setValidationState( state );
+    }
+
+    @Override
+    public void setTargetOrganizationalUnitGroupType( final ValidationState state ) {
+        targetOrganizationalUnitGroup.setValidationState( state );
+    }
+
+    @Override
+    public void setOrganizationalUnitEnabled( final boolean enabled ) {
+        sourceOrganizationalUnitTextBox.setEnabled( enabled );
+//        sourceOrganizationalUnitTextBox.refresh();
+    }
+
+    @Override
+    public void setForkEnabled( final boolean enabled ) {
+        fork.setEnabled( enabled );
+    }
+
+    @Override
+    public void setCancelEnabled( final boolean enabled ) {
+        cancel.setEnabled( enabled );
+    }
+
+    @Override
+    public void setPopupCloseVisible( final boolean closeVisible ) {
+        setClosable( closeVisible );
+    }
+
+    @Override
+    public void showBusyPopupMessage() {
+        BusyPopup.showMessage( CoreConstants.INSTANCE.Cloning() );
+    }
+
+    @Override
+    public void closeBusyPopup() {
+        BusyPopup.close();
+    }
+
+    @Override
+    public boolean showAgreeNormalizeNameWindow( final String normalizedName ) {
+        return Window.confirm( CoreConstants.INSTANCE.RepositoryNameInvalid() + " \"" + normalizedName + "\". " + CoreConstants.INSTANCE.DoYouAgree() );
+    }
+
+    @Override
+    public void alertRepositoryForked() {
+        Window.alert( CoreConstants.INSTANCE.RepoCloneSuccess() + "\n\n" + CommonConstants.INSTANCE.IndexClonedRepositoryWarning() );
```

Comment:
We should try to avoid use of ```Window.alert(...)``` and ```Window.confirm(...)``` as it's difficult to write Selenium tests for them - since different browsers implement these methods differently. Use of ```BaseModal``` and ```ModalFooterOKButton``` would be preferred (I realise some of this PR is copy and paste of old code.. but now is probably the best time to refactor).